README.md
specpile
========

specpile

sitemap/changelog.txt

ChangeLog for PHP XML Sitemap generator

http://www.xml-sitemaps.com
----------------------------------------------

2008-06-15 v2.9

   * automatic email notifications sent every time sitemap is generated
   * html sitemap customization: date format selection (xs_dateformat), order folders by name (xs_htmlnameorder)
   * http/1.1 protocol supported
   * memory usage optimized, new option to define memory limit
   * option to allow pages with http response code different from 200 to be included in sitemap (xs_allow_httpcode)
   * minor bug fixes



2007-12-27 v2.8

   * Live.com ping
   * meta description tag can be (optionally) extracted and inserted in ROR sitemap (advanced settings section)
   * Clear session (or others) parameters from URLs (advanced settings section)
   * IP address connection can be defined (advanced settings section)
   * Progress storage type: "serialize" or "var_export"
   * Changelog order can be changed in config file (xs_chlogorder option)


2007-10-21 v2.7

   * xml sitemap is now template based, which allows to integrate xsl stylesheet into your sitemap. Templates are defined in: sitemap_index_tpl.xml, sitemap_xml_tpl.xml and sitemap_ror_tpl.xml files
   * priority attribute can be assigned automatically by sitemap generator script now, just enable corresponding checkbox in configuration. Priority is descending with depth level: the highest priority is assigned to your homepage and the lower values are set for other pages, depending on how many "clicks" it takes to reach the page
   * minor bug fixes and improvements
   
   
2007-07-30 v2.6

	* sitemap ping for Ask.com and Moreover.com (Live Search)
	* memory usage optimized


2007-05-19 v2.5.3

	* titles are included in ROR sitemap now
	* minor bug fixes and improvements


2007-01-18 v2.5.2

	* minor bug fixes and improvements


2006-11-25 v2.5.1

	* sitemap protocol v0.9 is supported now


2006-11-24 v2.5

	* google sitemap ping URL has been updated
	* an option to automatically remove certain parameters from URLs has been implemented
	* user-agent can be customized now with xs_crawl_ident option in config file
	* robots.txt compliance can be turned on/off now with xs_robotstxt option in config file
	* regular expressions are allowed in "Individual parameters" option - just add a leading '$' sign in the line to enable it


2006-10-11 v2.4

	* yahoo auto notification option added
	* html sitemap is now fully customizable
	* minor bug fixes and improvements


2006-08-18 v2.3

	* ROR sitemaps are now supported
	* you can specify custom port for http connections now using xs_portconnection option in config.inc.php file
	* an option added to limit crawl depth level


2006-07-04 v2.2

	* the <meta name="robots" content="noindex, nofollow"> tag is supported now
	* xml sitemap size (max number of URLs per file) can be specified in config file 
	* location of text sitemap can be specified in config file
	* non-standard ports are supported like example.com:8080
	* minor bug fixes and improvements

2006-03-28 v2.1.1

	* minor bug fixes and improvements

2006-02-21 v2.1

    * html sitemap listing enhancements
	* minor bug fixes

2006-01-17 v2.0

	* HTML Site map generation is now included into the script. You are able to define the number of links per page in html sitemap and customize the page layout using simple template file.
	* Configuration screen interface improved
	* Mutiple minor changes


2005-11-14 v1.2

	* new feature: "Individual attributes" setting at Configuration screen. It allows you to set 
	  specific values for last modification time, frequency andpriority per page. 
	  To use it, define specific frequency and priority attributes in the following format:
	  "url substring,lastupdate YYYY-mm-dd,frequency,priority". 
	* new feature: "Include ONLY URLs" setting. This setting is the opposite to "Exclude URLs". When 
	  it is not empty, ONLY the urls that match the substring entered are included into sitemap.
	* configuration page interface improved (parameters are included into two groups now: main parameters and optional settings)


2005-10-27 v1.1.3

	* the generator creates a sitemap in TEXT FORMAT for you (one URL per line). You can download it from the "Sitemap details" block at the left column of the page. This URLs list can be submitted to Yahoo! at http://submit.search.yahoo.com/free/request


2005-09-14 v1.1.2

	* (fix) asp sites crawling problem


2005-08-30 v1.1.1

	* "sleep" delays are now available to configure to reduce the load on your server
	* new configuration directive: "Do not parse URLs" allows to skip fetching some pages to boost
	  the speed of sitemap generation of the large sites
	* single quoted <base> tag


2005-08-08 v1.1.0

	* "resume the interrupted session" feature added
	* urls extraction has been improved (some uncatched urls are now processed)
	* performance/memory usage optimization


2005-07-31 v1.08

	* generator pages access can be restricted
	* minor improvements


2005-07-20 v1.07

	* CRLF problem with sitemaps (that may lead to a parse problem in rare cases) fixed
	* script configuration auto-filled with initial values	
	* robots.txt is now processed both from top level and from the starting URL directory


2005-07-14 v1.06

	* support of (') character in URLs


2005-07-14 v1.05

	* "Content-type" header is checked and only "text/html" entries are processed
	* "extensions" config settings trailing space problem fixed
	* "gz" added to default excluded extensions
	* "change_frequency" setting fix
	* skip redirections to non-html files 
	* interface enhacements


2005-07-07 v1.04

	* full config with initial settings included


2005-07-07 v1.03

	* "custom lastmod time" bug fixed
	* all pages exluded when "exclude extensions" field is empty


2005-07-05 v1.02

	* robots.txt protocol is supported ('*' and 'googlebot' user-agents are checked)
	* view current process state when crawler is in background mode
	* interrupt the crawler in background mode

sitemap/config.inc.php
<?php
$grab_parameters = array(
'xs_inc_skip' => '\\.(pdf|doc|txt|rtf)',
'xs_exc_skip' => '\\.(zip|m4v|rar|tar|bz2|tgz|exe|gif|jpg|png|class|jar|mpeg|mpg|mp3|wav|mp4|avi|wmv|gz|mov|mid|ra|ram)',
'xs_proto_skip' => '(\\#|mms:|mailto:|https:|javascript:|ftp:|news:|aim:)',
'xs_exec_time' => '90000',
'xs_cache' => '',
'xs_max_pages' => '0',
'xs_initurl' => 'http://specpile.com',
'xs_freq' => 'daily',
'xs_lastmod' => '1',
'xs_lastmodtime' => '2005-06-30 19:35:14',
'xs_priority' => '0.5',
'xs_descpriority' => '0.8',
'xs_autopriority' => '1',
'xs_smname' => 'C:\\wamp\\www/sitemap.xml',
'xs_smurl' => 'http://localhost\\/sitemap.xml',
'xs_gping' => '1',
'xs_yping' => '',
'xs_makehtml' => '1',
'xs_maketxt' => '1',
'xs_makeror' => '1',
'xs_makebase' => '0',
'xs_compress' => '0',
'xs_excl_urls' => '',
'xs_notconfigured' => '0',
'xs_login' => '',
'xs_password' => '',
'xs_savestate_time' => '180',
'xs_ipconnection' => '',
'xs_portconnection' => '',
'xs_sm_size' => '40000',
'xs_sm_text_filename' => '',
'xs_sm_text_url' => '',
'xs_cleanurls' => '',
'xs_usecurl' => '0',
'xs_robotstxt' => '1',
'xs_dumptype' => 'serialize',
'xs_cleanpar' => '',
'xs_chlogorder' => 'asc',
'xs_exclude_check' => '1',
'xs_dateformat' => 'Y, F j',
'xs_allow_httpcode' => '',
'xs_chlog' => '1',
'xs_htmlname' => 'C:\\wamp\\www\\sitemap/data/sitemap.html',
'xs_htmlpart' => '1000',
'htmlurl' => 'http://localhost/sitemap/data/sitemap.html',
'xs_max_depth' => '',
'xs_memlimit' => '',
'xs_delay_req' => '',
'xs_delay_ms' => '',
'xs_incl_urls' => '',
'xs_incl_only' => '',
'xs_ind_attr' => '',
'xs_email' => '',
'xs_metadesc' => '0',);
?>